### PR TITLE
added a utility function to get the log of a list of doubles.

### PR DESCRIPTION
**Pull request by yfarjoun**
- added a utility function to get the log of a list of doubles
- exposed TestNGUtil in picard-public
- added tests for new functionality

### CODE REVIEW

File: src/java/picard/util/MathUtil.java [hunk 1]
```diff
@@ -325,6 +325,15 @@ public double getLogValue(final double nonLogValue) {
             return Math.log(nonLogValue) / log_of_base;
         }
 
+        /** Returns the log-representation of the provided decimal array. */
+        public double[] getLogValue(final double[] nonLogArray) {
```
Comment (akiezun) [inline src/java/picard/util/MathUtil.java]: static

File: src/java/picard/util/TestNGUtil.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import java.util.List;
+
+import org.testng.Assert;
+
+import static java.lang.Math.abs;
+
+/**
+ * Small class implementing some utility functions that are useful for test and interfacing with the TestNG framework.
+ */
+public class TestNGUtil {
```
Comment (akiezun) [inline src/java/picard/util/TestNGUtil.java]: make final, add private constructor

File: src/java/picard/util/TestNGUtil.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            parameterList.add(i.next().toObjectArray());
+        }
+        Object[][] parameterArray = new Object[parameterList.size()][];
+        for (int j = 0; j < parameterList.size(); j++) {
+            parameterArray[j] = parameterList.get(j);
+        }
+        return parameterArray;
+    }
+
+    static final double EPSILON = 1e-300; //a constant near the smallest possible positive value representable in double.
```
Comment (akiezun) [inline src/java/picard/util/TestNGUtil.java]: move to top of class

**Comment by kbergin** [comment]
:+1: All looks good to me. I checked and I see that the TestNG is the exact same as the private, and everything else looks good!

File: src/java/picard/util/TestNGUtil.java (hunk 3 above)
Comment (akiezun) [inline src/java/picard/util/TestNGUtil.java]: if you want the actual smallest number, use this `Double.longBitsToDouble(0x1L)`. It's `4.9E-324`

File: src/java/picard/util/TestNGUtil.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+    /**
+     * Small utility function for determining if two doubles are within a _relative_ accuracy of each other.
+     *
+     * @param lhs first number
+     * @param rhs second number
+     * @param accuracy maximal allowed relative difference between the two numbers
+     * @return true if numbers are within the relative tolerance of each other
+     */
+    static public boolean compareDoubleWithAccuracy(final double lhs, final double rhs, final double accuracy) {
```
Comment (akiezun) [inline src/java/picard/util/TestNGUtil.java]: proper order is `static public boolean`

File: src/java/picard/util/TestNGUtil.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        if (accuracy <= 0) throw new IllegalArgumentException("Accuracy must be positive.");
+        return abs(lhs - rhs) / (abs(lhs) + abs(rhs) + EPSILON) < accuracy;
+    }
+
+    static public void assertEqualDoubleArrays(final double[] lhs, final double[] rhs, final double accuracy) {
```
Comment (akiezun) [inline src/java/picard/util/TestNGUtil.java]: `public static boolean`

Comment (akiezun) [inline src/java/picard/util/TestNGUtil.java]: check for null arguments on all arrays

File: src/java/picard/util/TestNGUtil.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+import org.testng.Assert;
+
+import static java.lang.Math.abs;
+
+/**
+ * Small class implementing some utility functions that are useful for test and interfacing with the TestNG framework.
+ */
+public class TestNGUtil {
+    public static abstract class SingleTestUnitTest<TESTCASE_TYPE extends TestNGParameterizable> {
```
Comment (akiezun) [inline src/java/picard/util/TestNGUtil.java]: Convention is to use single letter for formal type parameters or use class name conventions: http://google.github.io/styleguide/javaguide.html#s5.2.8-type-variable-names

**Comment by akiezun** [comment]
small things to fix. merge at will after that

File: src/java/picard/util/MathUtil.java (hunk 1 above)
Comment (yfarjoun) [inline src/java/picard/util/MathUtil.java]: no, it depends on the base

**Comment by yfarjoun** [comment]
responded to all but one comment. 
waiting for tests.

**Pull request merged**